File: Medium/2026-06-21-1833-Maximum-Ice-Cream-Bars/solution.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        count = [0] * 100001
        
        for cost in costs:
            count[cost] += 1
        
        total_bars = 0
        for price in range(1, 100001):
            if count[price] == 0:
                continue
            if coins >= price:
                can_buy = min(count[price], coins // price)
                total_bars += can_buy
                coins -= can_buy * price
            else:
                break
        
        return total_bars

File: Medium/2026-06-21-1833-Maximum-Ice-Cream-Bars/solution.java
import java.util.Arrays;

class Solution {
    public int maxIceCream(int[] costs, int coins) {
        int[] count = new int[100001];
        for (int cost : costs) {
            count[cost]++;
        }
        
        int totalBars = 0;
        for (int price = 1; price < count.length; price++) {
            while (count[price] > 0 && coins >= price) {
                coins -= price;
                totalBars++;
                count[price]--;
            }
        }
        
        return totalBars;
    }
}